test/helpers/kubectl.go

```diff
@@ -476,19 +476,23 @@ func (kub *Kubectl) WaitForKubeDNSEntry(name string) error {
 	}
 	// https://bugs.launchpad.net/ubuntu/+source/bind9/+bug/854705
 	digCMD := "dig +short %s @%s | grep -v -e '^$'"
+	digCMDFallback := "dig +tcp %s @%s"
 
-	podsIPs, err := kub.GetPodsIPs(KubeSystemNamespace, "k8s-app=kube-dns")
+	host, _, err := kub.GetServiceHostPort(KubeSystemNamespace, "kube-dns")
 	if err != nil {
-		logger.WithError(err).Error("cannot get kube-dns pods")
+		logger.WithError(err).Error("cannot get kube-dns service IP")
 		return err
 	}
 
 	body := func() bool {
 		var result bool
-		for _, ip := range podsIPs {
-			res := kub.Exec(fmt.Sprintf(digCMD, name, ip))
-			result = res.WasSuccessful()
+		res := kub.Exec(fmt.Sprintf(digCMD, name, host))
+		if !res.WasSuccessful() {
+			// If it fails we want to know if it's because of connection
+			// cannot be established or DNS does not exist.
+			_ = kub.Exec(fmt.Sprintf(digCMDFallback, name, host))
 		}
+		result = res.WasSuccessful()
 		return result
```

Comment:
If you wanted to get rid of the result variable, we can return `res.WasSuccessful()` directly but it's not a big deal

Reply:
+1